waku/v2/protocol/app_design.go

```diff
@@ -0,0 +1,82 @@
+package protocol
+
+import (
+	"context"
+	"errors"
+	"sync"
+)
+
+type AppDesign struct {
```

Comment:
Maybe instead of `AppDesign`, it should be `Service` or `WakuProtocol` or similar? This is used to control the lifetime of the protocols, so probably should have a name closely associated to its purpose

Reply:
Service may cause confusion with Service-Protocol terminology in Waku which doesn't include relay. 
Maybe we can just call it Protocol or WakuProtocol.